src/Error/Error.php

```diff
@@ -114,15 +115,15 @@ public function __construct(
         $this->source     = $source;
         $this->positions  = $positions;
         $this->path       = $path;
-        $this->extensions = $extensions ?: (
+        $this->extensions = count($extensions) > 0 ? $extensions : (
         $previous && $previous instanceof self
             ? $previous->extensions
             : []
         );
 
         if ($previous instanceof ClientAware) {
             $this->isClientSafe = $previous->isClientSafe();
-            $this->category     = $previous->getCategory() ?: self::CATEGORY_INTERNAL;
+            $this->category     = $previous->getCategory() !== '' ? $previous->getCategory() : self::CATEGORY_INTERNAL;
```

Comment:
Can we invert the condition to avoid negation?

Reply:
Changed in 226b0182ffe4578bded7061cd737bb38fdb0db98